### ISSUE TITLE
Helix instance freezes with no output

### ISSUE THREAD

**Issue by stevengill** [adobe/helix-cli#856]
**Description**
Alright, this is a very weird issue but I do have reproduction steps. We are working on support for a strain that points to a subdirectory in git. This works fine, we can load files that are in the root of that subdirectory. Any files that are nested deeper though are causing this freeze.

Content repo: https://github.com/abhishekchandran/contributor.en

**To Reproduce**
1. `git clone` our [developer.adobe.com](https://github.com/adobe/developer.adobe.com) repo and `git checkout TocSupport` branch. (This [PR](https://github.com/adobe/developer.adobe.com/pull/131) represents changes on TocSupport from master )
2. `npm install` and then `npm start -- --log-level=debug`
3. Go to `http://localhost:3000/contributor/docs`, loads fine (this is loading [this strain](https://github.com/adobe/developer.adobe.com/blob/TocSupport/helix-config.yaml#L52-L57))
4. Click the [Using Markdown](http://localhost:3000/contributor/docs/writing-essentials/using-markdown.html) option in the left nav. You will see it never loads and helix instance freezes without the proper output of why.

**Related Note* If you click on any of the other nested elements in the left nav, you get a different error. `Error: Found ambigous frontmatter fence: No empty line after the block! Make sure your frontmatter blocks contain no empty lines and your horizontal rules have an empty line before AND after them.` Adding a newline to the markdown file before and after the frontmatter fixes this (we did this already to using-markdown.md). Once frontmatter is fixed, it does the error I described in this issue. The weird thing is that this frontmatter loads fine for the files that are in the root of the subdirectory (without the newlines). This parsing error only occurs for the files deeper in the subdirectory.

**Expected behavior**
It is expected to load and render the markdown.

**Version:**
2.1.6

**Additional context**
here is my stack trace
```
[hlx] debug: current strain: contributor-docs-staging: {
  "code": "https://github.com/adobe/developer.adobe.com.git#master",
  "content": "https://github.com/abhishekchandran/contributor.en.git/help#master",
  "static": {
    "protocol": "http",
    "host": "127.0.0.1:60488",
    "port": 60488,
    "hostname": "127.0.0.1",
    "owner": "helix",
    "repo": "github.com--adobe--developer.adobe.com",
    "ref": "TocSupport",
    "path": "/htdocs"
  },
  "directoryIndex": "introduction.html",
  "package": "developer-adobe-com/633d4dd15734e619449fd1b121d6e97d5a21b6ce",
  "name": "contributor-docs-staging",
  "urls": [
    "https://adobedevsite.helix-demo.xyz/contributor/docs"
  ],
  "url": "https://adobedevsite.helix-demo.xyz/contributor/docs"
}
[hlx] debug: resolved /contributor/docs/writing-essentials/using-markdown.html -> html
[hlx] debug: Constructing HTML Pipeline
[hlx] debug: Creating pipeline
[hlx] debug: Coercing secrets
[hlx] debug: Running HTML pipeline
[hlx] debug: fetching Markdown from https://raw.github.com/abhishekchandran/contributor.en/master/help/writing-essentials/using-markdown.md
[hlx] debug: Parsing markdown from request body starting with
[hlx] warn: Whitelist forbids embedding of URL: https://jmalarcon.github.io/markdowntables/
[hlx] warn: Whitelist forbids embedding of URL: https://daringfireball.net/projects/markdown/syntax
[hlx] warn: Whitelist forbids embedding of URL: https://help.github.com/articles/markdown-basics/
[hlx] debug: Parsing Markdown Metadata from 1 sections
[hlx] debug: Turning Markdown into HTML from object
[hlx] debug: Making images responsive in object
[hlx] debug: Making image assets/no-localize/adobe_standard_logo.png responsive
[hlx] debug: Making image assets/no-localize/adobe_standard_logo.png responsive
[hlx] debug: request url: /contributor/docs/writing-essentials/using-markdown.html
[hlx] debug: request path: /contributor/docs/writing-essentials/using-markdown.html
[hlx] debug: action path: /help/writing-essentials/using-markdown.md
[hlx] debug: action rootPath: /contributor/docs
[hlx] debug: strain header: contributor-docs-staging
[hlx] debug: dispatch url: /contributor/docs/writing-essentials/using-markdown.docs.html
[hlx] debug: Setting content type header to text/html
[hlx] debug: Setting Surrogate-Key header
[hlx] debug: current strain: contributor-docs-staging: {
  "code": "https://github.com/adobe/developer.adobe.com.git#master",
  "content": "https://github.com/abhishekchandran/contributor.en.git/help#master",
  "static": {
    "protocol": "http",
    "host": "127.0.0.1:60488",
    "port": 60488,
    "hostname": "127.0.0.1",
    "owner": "helix",
    "repo": "github.com--adobe--developer.adobe.com",
    "ref": "TocSupport",
    "path": "/htdocs"
  },
  "directoryIndex": "introduction.html",
  "package": "developer-adobe-com/633d4dd15734e619449fd1b121d6e97d5a21b6ce",
  "name": "contributor-docs-staging",
  "urls": [
    "https://adobedevsite.helix-demo.xyz/contributor/docs"
  ],
  "url": "https://adobedevsite.helix-demo.xyz/contributor/docs"
}
[hlx] debug: resolved /contributor/docs/writing-essentials/using-markdown.docs.html -> docs_html
[hlx] debug: Constructing HTML Pipeline
[hlx] debug: Creating pipeline
[hlx] debug: Coercing secrets
[hlx] debug: Running HTML pipeline
[hlx] debug: fetching Markdown from https://raw.github.com/abhishekchandran/contributor.en/master/help/writing-essentials/using-markdown.md
[hlx] debug: Parsing markdown from request body starting with
[hlx] warn: Whitelist forbids embedding of URL: https://jmalarcon.github.io/markdowntables/
[hlx] warn: Whitelist forbids embedding of URL: https://daringfireball.net/projects/markdown/syntax
[hlx] warn: Whitelist forbids embedding of URL: https://help.github.com/articles/markdown-basics/
[hlx] debug: Parsing Markdown Metadata from 1 sections
[hlx] debug: Turning Markdown into HTML from object
[hlx] debug: Making images responsive in object
[hlx] debug: Making image assets/no-localize/adobe_standard_logo.png responsive
[hlx] debug: Making image assets/no-localize/adobe_standard_logo.png responsive
[hlx] debug: docs_html.pre.js - Fetching the commits history
[hlx] debug: docs_html.pre.js - Fetching... https://api.github.com/repos/abhishekchandran/contributor.en/commits?path=/help/writing-essentials/using-markdown.md&sha=master
[hlx] debug: docs_html.pre.js - Extracting committers from metadata
[hlx] debug: docs_html.pre.js - Number of committers extracted: 3
[hlx] debug: docs_html.pre.js - Extracting last modified from metadata
[hlx] debug: docs_html.pre.js - Managed to fetch a last modified: Thu May 09 2019 11:08:22 GMT-0700 (Pacific Daylight Time)
[hlx] debug: docs_html.pre.js - Assembling edit URL
[hlx] debug: docs_html.pre.js - Creating TOC
[hlx] debug: docs_html.pre.js - VDOM children processed (stripping leading title)
[hlx] debug: docs_html.pre.js - Fetching the nav
[hlx] debug: docs_html.pre.js - Fetching... https://api.github.com/repos/abhishekchandran/contributor.en/git/trees/master?recursive=1
[hlx] debug: docs_html.pre.js - Development path to valid md files to generate nav: /contributor/docs/TOC
[hlx] debug: Setting content type header to text/html
[hlx] debug: Setting Surrogate-Key header
[hlx] debug: current strain: contributor-docs-staging: {
  "code": "https://github.com/adobe/developer.adobe.com.git#master",
  "content": "https://github.com/abhishekchandran/contributor.en.git/help#master",
  "static": {
    "protocol": "http",
    "host": "127.0.0.1:60488",
    "port": 60488,
    "hostname": "127.0.0.1",
    "owner": "helix",
    "repo": "github.com--adobe--developer.adobe.com",
    "ref": "TocSupport",
    "path": "/htdocs"
  },
  "directoryIndex": "introduction.html",
  "package": "developer-adobe-com/633d4dd15734e619449fd1b121d6e97d5a21b6ce",
  "name": "contributor-docs-staging",
  "urls": [
    "https://adobedevsite.helix-demo.xyz/contributor/docs"
  ],
  "url": "https://adobedevsite.helix-demo.xyz/contributor/docs"
}
[hlx] debug: resolved /contributor/docs/TOC.summary.html -> summary_html
[hlx] debug: Constructing HTML Pipeline
[hlx] debug: Creating pipeline
[hlx] debug: Coercing secrets
[hlx] debug: Running HTML pipeline
[hlx] debug: fetching Markdown from https://raw.github.com/abhishekchandran/contributor.en/master/help/TOC.md
[hlx] debug: Parsing markdown from request body starting with ---
[hlx] debug: Parsing Markdown Metadata from 1 sections
[hlx] debug: Turning Markdown into HTML from object
[hlx] debug: Making images responsive in object
[hlx] debug: summary_html.pre.js - Requested path: /help/TOC.md
[hlx] debug: summary_html.pre.js - Extracting nav
[hlx] debug: summary_html.pre.js - Managed to collect some content for the nav: 1
[hlx] debug: Setting content type header to text/html
[hlx] debug: Setting Surrogate-Key header
```

**Comment by trieloff** [comment]
Doesn't seem to be a pipeline/parsing issue per se, as it works fine with the default pipeline and a different template.

**Comment by stevengill** [comment]
Hmm, you think it might be related to our code then. Okay I will give a stab at debugging this again on our side.

**Comment by tripodsan** [comment]
endless loop is caused by a bug in the esi processor:
https://github.com/adobe/helix-simulator/issues/212#issuecomment-490381202

**Comment by trieloff** [comment]
@tripodsan given that we go through the HAST anyway, would it make sense to throw error messages in the log when unknown ESI tags are used? Or even fail entirely?

**Comment by tripodsan** [comment]
> given that we go through the HAST anyway, 

well, if we just use the nodesi parser, then it will be hard...

> would it make sense to throw error messages in the log when unknown ESI tags are used? Or even fail entirely?

I think issuing a warning would be good....

**Comment by tripodsan** [comment]
fixed with #873
